packages/python-packages/api-stub-generator/apistub/_apiview.py

```diff
@@ -13,7 +13,8 @@
 JSON_FIELDS = ["Name", "Version", "VersionString", "Navigation", "Tokens", "Diagnostics"]
 
 HEADER_TEXT = "# Package is parsed using api-stub-generator(version:{})".format(VERSION)
-COMPLEX_DATA_TYPE_PATTERN = re.compile("([\w.]+)(\[|\()[^\]\)]+(\]|\))")
+TYPE_NAME_REGEX = re.compile("(~?[a-zA-Z\d.]+)")
+TYPE_OR_SEPERATOR = " or "
```

Comment:
Could potentially add another separator for ", ", just in case we see something like:
`:type: list, str or int`